pkg/native/serializers/serializer_cdx.go

```diff
@@ -368,17 +423,26 @@ func (s *SerializerCDX) nodeToComponent(n *sbom.Node) *cdx.Component {
 	return c
 }
 
-// renderVersion calls the official CDX serializer to render the BOM into a
-// specific version
-func (s *SerializerCDX) renderVersion(cdxVersion cdx.SpecVersion, doc interface{}, wr io.Writer) error {
+// Render calls the official CDX serializer to render the BOM into a specific version
+func (s *CDX) Render(doc interface{}, wr io.Writer, o *native.RenderOptions) error {
 	if doc == nil {
 		return errors.New("document is nil")
 	}
 
-	encoder := cdx.NewBOMEncoder(wr, cdx.BOMFileFormatJSON)
+	version, err := cdxformats.ParseVersion(s.version)
+	if err != nil {
+		return fmt.Errorf("getting CDX version: %w", err)
+	}
+
+	encoding, err := cdxformats.ParseEncoding(s.encoding)
+	if err != nil {
+		return fmt.Errorf("getting CDX encoding: %w", err)
+	}
+
+	encoder := cdx.NewBOMEncoder(wr, encoding)
 	encoder.SetPretty(true)
 
-	if err := encoder.EncodeVersion(doc.(*cdx.BOM), cdxVersion); err != nil {
+	if err := encoder.EncodeVersion(doc.(*cdx.BOM), version); err != nil {
```

Comment:
We need to test if this interface can be cast, otherwise, we could introduce a panic here:
```suggestion
	if _, ok := doc.(*cdx.BOM); !ok {
		return errors.New("document is not a cyclonedx bom")
	}
	if err := encoder.EncodeVersion(doc.(*cdx.BOM), version); err != nil {
```